### PR TITLE
Handle nested arrays and objects in responses

### LINKED ISSUE #7: How to support arrays in behavioural testing
Given the test data:

```json
{
  "root": "text",
  "other": "field",
  "child": { "key": "stuff" },
  "comments": [
    {
      "field": "things",
      "second": 1,
      "ignored": true,
      "common": 12
    },
    {
      "field": "different",
      "second": 2,
      "common": 12
    }
  ]
}
```

We should be able to say one of the elements in the comments matches a given set of key/values

- Needs to support single object responses and list responses

## Option 1 in attributes

```gherkin
Then the response has the following attributes:
    | attribute          | type    | value     |
    | root               | string  | text      |
    | child : key        | string  | stuff     |
    | comment #1: field  | string  | things    |
    | comment #1: second | integer | 1         |
    | comment #1: common | integer | 12        |
    | comment #2: field  | string  | different |
    | comment #2: common | integer | 12        |
```

- The `#1` / `#2` notation just signifies that this element is grouped together and does not denote order [confusing?]
- Requries a custom search:
  - The search should be lazy and of limited scope (as soon as it finds any element in the list that matches all the keys provided for that entry it is removed from the search)
  - The child matching might be a np complete problem?
    - If there is a child that matches 2 selectors which one do we remove off the search

## Option 2 using words

```gherkin
Then the response has|contains a list of comments // response contains {"comments":[]}
Then the response has|contains a list of 2 comments // response contains {"comments":[{},{}]}
Then the response has|contains a child with attributes: // response has object comment with object attributes
    | field | string | things |
    | second | integer | 1 |
Then the response has|contains one comment with attributes: // response has list comments with a single entry matching the provided attributes
    | field | string | different |
Then the response has|contains two comments with attributes: // response has list comments with 2 matching the provided attributes
    | common | integer | 12 |
```

- Only deals with single level depth (how far should we go?!)
- Can use current object matching techniques
- How to we find 2 top level objects that have items with children?
    ```gherkin
    Then three posts contains two comments with attributes:
    ```
- ... but what about matching top level and child elements? (not sure)

### PR DESCRIPTION
**Pull request by h-bragg**
Resolves #7 

Able to handle nested arrays and objects

### CODE REVIEW

**Comment by h-bragg** [comment]
Added support for number modifiers everywhere:

```gherkin
Then at least two posts have more than 3 comments with attributes:
     | attribute | type | value |
     | foo | string | bar |
```

**Comment by h-bragg** [comment]
@james-weaver (sorry, mind re-reviewing)

**Comment by h-bragg** [comment]
(hopefully the last change)

File: lib/cucumber-rest-bdd/steps/response.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    expected = get_attributes(attributes.hashes)
+    groups = nesting
+    grouping = get_grouping(groups)
+    grouping.push({
+        root: true,
+        type: 'multiple',
+        count: count.to_i,
+        count_mod: count_mod
+    })
+    puts 'count_mod', count_mod
```
Comment (james-weaver) [inline lib/cucumber-rest-bdd/steps/response.rb]: Unnecessary `puts`

File: lib/cucumber-rest-bdd/steps/response.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+Then(/^(#{FEWER_MORE_THAN})?\s*(#{CAPTURE_INT}|\d+) (?:.*?) ((?:(?:has|have|having|contain|contains|containing|with) (?:a|an|(?:(?:#{FEWER_MORE_THAN})?\s*#{CAPTURE_INT}|\d+)) (?:\w+)\s?)+)$/) do |count_mod, count, nesting|
+    groups = nesting
+    grouping = get_grouping(groups)
+    grouping.push({
+        root: true,
+        type: 'multiple',
+        count: count.to_i,
+        count_mod: count_mod
+    })
+    puts 'count_mod', count_mod
```
Comment (james-weaver) [inline lib/cucumber-rest-bdd/steps/response.rb]: Unnecessary `puts`

File: STEPS.md [hunk 3]
```diff
@@ -115,23 +115,84 @@ Then the response is a list of 12 items             Then the JSON response shoul
                                                       entries
 
 Then the response is a list of at least 12 items    <N/A>
```
Comment (james-weaver) [inline STEPS.md]: Is this not supported in functional tests?

**Pull request merged**